crates/vision/src/image_receiver.rs

```diff
@@ -1,20 +1,27 @@
+use std::time::{Duration, SystemTime};
+
 use color_eyre::Result;
 use context_attribute::context;
-use framework::MainOutput;
-use hardware::CameraInterface;
+use framework::{AdditionalOutput, MainOutput};
+use hardware::{CameraInterface, TimeInterface};
 use serde::{Deserialize, Serialize};
 use types::{camera_position::CameraPosition, ycbcr422_image::YCbCr422Image};
 
 #[derive(Deserialize, Serialize)]
-pub struct ImageReceiver {}
+pub struct ImageReceiver {
+    last_cycle_start: SystemTime,
+}
 
 #[context]
-pub struct CreationContext {}
+pub struct CreationContext {
+    hardware_interface: HardwareInterface,
+}
 
 #[context]
 pub struct CycleContext {
     hardware_interface: HardwareInterface,
     camera_position: Parameter<CameraPosition, "image_receiver.$cycler_instance.camera_position">,
+    cycle_time: AdditionalOutput<Duration, "cycle_time">,
```

Comment:
naming... what do you think of `last_cycle_duration`

Reply:
Done